### ISSUE TITLE
Syntax highlighting for markdown YAML front matter

### ISSUE THREAD

**Issue by aloreaux** [ScintillaOrg/lexilla#59]
Is it possible to get markdown YAML front matter syntax highlighting added? It currently is not working as it should be. None of the keys get highlighted, everything is default

**Comment by nyamatongwe** [comment]
aloreaux:

> Is it possible to get markdown YAML front matter syntax highlighting added? It currently is not working as it should be. None of the keys get highlighted, everything is default

   Lexilla doesn’t support combining lexers - you’d have to add YAML interpretation code into the Markdown lexer.

   Neil

**Comment by nyamatongwe** [comment]
Some documentation in [frontmatter](https://www.markdownlang.com/advanced/frontmatter.html).

**Comment by PGTBoos** [comment]
Is there a specific reason why Lexilla doesn’t support combining lexers?
As these these it's quite common for markdown, to put text and code in the same file.
I just wonder what it would take to get this done, and perhaps would start coding such a future in spare time. (not a hard promise).

**Comment by nyamatongwe** [comment]
> Is there a specific reason why Lexilla doesn’t support combining lexers?

There are multiple factors like the limits on styles, determining language boundaries and transitions, and merging features like keyword lists. I have tried to implement combining lexers a couple of times (see [SinkWorld](https://www.scintilla.org/SinkWorld.html)) and didn't produce a great result.

Maybe someone else will be sufficiently interested to design and implement this.

**Comment by PGTBoos** [comment]
Yes well I rewrote it the modern way test runs all fine with 2 exceptions for out of tokens .. the main thing though it requires a huge rewrite I have no endless tokens I monitored it all and test runs preserved multiple languages but I'm out of tokens and have real work as well.

Verzonden vanaf Outlook voor Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by nyamatongwe** [comment]
Currently, I will not accept AI-generated code into this project so please do not propose any such code.

Scintilla communicates with lexers over the [Lexilla protocol](https://www.scintilla.org/LexillaDoc.html) which is open to multiple implementations. If you want to publish your own project that uses AI-generated code then you can make a new project that implements this protocol similar to [Scintillua](https://github.com/orbitalquark/scintillua). There is also a simple example in `lexilla/examples/SimpleLexer`.

**Comment by PGTBoos** [comment]
Well only a redesign could fix it, so I'm out

Verzonden vanaf Outlook voor Android<https://aka.ms/AAb9ysg>
________________________________
